x-pack/plugins/security_solution/public/actions/register.ts

```diff
@@ -7,14 +7,89 @@
 
 import { CELL_VALUE_TRIGGER } from '@kbn/embeddable-plugin/public';
 import type { UiActionsStart } from '@kbn/ui-actions-plugin/public';
+import type * as H from 'history';
 import type { SecurityAppStore } from '../common/store/types';
-import { AddToTimelineAction } from './add_to_timeline';
-import { CopyToClipboardAction } from './copy_to_clipboard';
+import type { StartPlugins, StartServices } from '../types';
+import { EmbeddableCopyToClipboardAction } from './copy_to_clipboard/embeddable_copy_to_clipboard_action';
+import { createFilterInAction } from './filter/filter_in';
+import { createFilterOutAction } from './filter/filter_out';
+import { createAddToTimelineAction, EmbeddableAddToTimelineAction } from './add_to_timeline';
+import { createShowTopNAction } from './show_top_n/show_top_n';
+import { SECURITY_SOLUTION_ACTION_TRIGGER, TIMELINE_ACTION_TRIGGER } from '../../common/constants';
+import { createCopyToClipboardAction } from './copy_to_clipboard/copy_to_clipboard';
+import { createTimelineFilterInAction, createTimelineFilterOutAction } from './filter';
 
-export const registerActions = (uiActions: UiActionsStart, store: SecurityAppStore) => {
-  const addToTimelineAction = new AddToTimelineAction(store);
+export const registerActions = (
+  plugins: StartPlugins,
+  store: SecurityAppStore,
+  history: H.History,
+  services: StartServices
+) => {
+  registerEmbeddableActions(plugins.uiActions, store);
+  registerUiActions(plugins.uiActions, store, history, services);
+  registerTimelineUiActions(plugins.uiActions, store, history, services);
+};
+
+const registerEmbeddableActions = (uiActions: UiActionsStart, store: SecurityAppStore) => {
+  const addToTimelineAction = new EmbeddableAddToTimelineAction(store);
   uiActions.addTriggerAction(CELL_VALUE_TRIGGER, addToTimelineAction);
 
-  const copyToClipboardAction = new CopyToClipboardAction();
+  const copyToClipboardAction = new EmbeddableCopyToClipboardAction();
   uiActions.addTriggerAction(CELL_VALUE_TRIGGER, copyToClipboardAction);
 };
+
+const registerUiActions = (
+  uiActions: UiActionsStart,
+  store: SecurityAppStore,
+  history: H.History,
+  services: StartServices
+) => {
+  const filterInAction = createFilterInAction({
```

Comment:
I like this way to create the actions, passing the order by parameter. Could we create the Lens cell actions the same way?